File: java/src/main/java/deltix/luminary/LiteralDate.java
package deltix.luminary;

public class LiteralDate extends Literal {
    private String value;

    public LiteralDate(String value) {
        super(LiteralKind.DATE);
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    @Override
    public String toString() {
        return "\"" + value + "\"";
    }

    public static final String MIN_VALUE_AS_STRING = "#MIN";
    public static final String MAX_VALUE_AS_STRING = "#MAX";
}

File: csharp/main/LiteralDate.cs
using System;

namespace Deltix.Luminary
{
	public sealed class LiteralDate : Literal, ILiteral<String>
	{
		public const String MinValueAsString = "#MIN";
		public const String MaxValueAsString = "#MAX";

		public LiteralDate(String value)
			: base(LiteralKind.Date)
		{
			Value = value;
		}

		public String Value { get; }

		public override String ToString()
		{
			return $"\"{Value}\"";
		}
	}
}
